modular_skyrat/master_files/code/modules/client/preferences/clothing.dm

```diff
@@ -35,3 +35,58 @@
 	var/species_type = preferences.read_preference(/datum/preference/choiced/species)
 	var/datum/species/species = new species_type
 	return !(TRAIT_NO_UNDERWEAR in species.inherent_traits)
+
+/datum/preference/choiced/bra
+	savefile_key = "bra"
+	savefile_identifier = PREFERENCE_CHARACTER
+	main_feature_name = "Bra"
+	category = PREFERENCE_CATEGORY_CLOTHING
+	should_generate_icons = TRUE
+
+/datum/preference/choiced/bra/init_possible_values()
+	return assoc_to_keys_features(GLOB.bra_list)
+
+/datum/preference/choiced/bra/icon_for(value)
+	var/static/icon/body
+	if (isnull(body))
+		body = icon('icons/mob/human/bodyparts_greyscale.dmi', "human_r_arm")
+		body.Blend(icon('icons/mob/human/bodyparts_greyscale.dmi', "human_l_arm"), ICON_OVERLAY)
+		body.Blend(icon('icons/mob/human/bodyparts_greyscale.dmi', "human_r_hand"), ICON_OVERLAY)
+		body.Blend(icon('icons/mob/human/bodyparts_greyscale.dmi', "human_l_hand"), ICON_OVERLAY)
+		body.Blend(icon('icons/mob/human/bodyparts_greyscale.dmi', "human_chest_m"), ICON_OVERLAY)
+
+	var/icon/icon_with_bra = icon(body)
+
+	if (value != "Nude")
+		var/datum/sprite_accessory/accessory = GLOB.bra_list[value]
+		icon_with_bra.Blend(icon(accessory.icon, accessory.icon_state), ICON_OVERLAY)
+
+	icon_with_bra.Crop(10, 11, 22, 23) // SKYRAT EDIT CHANGE : ORIGINAL - icon_with_undershirt.Crop(9, 9, 23, 23)
+	icon_with_bra.Scale(32, 32)
+	return icon_with_bra
```

Comment:
For this part of the proc, please refer to how we did it with `generate_underwear_icon()`, because I foresee this causing potential issues with colorations and offsets, potentially.